File: problems/leetcode/RemoveDuplicatesFromSortedArray.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int newIndex = 1;

        for (int i = 0 ; i < nums.length;) {
            int j = i + 1;
            while (j < nums.length && nums[i] == nums[j]) j++;
            if (j == nums.length) break;

            nums[newIndex++] = nums[j];
            i = j;
        }

        return newIndex;
    }
}

File: problems/leetcode/remove-duplicates-from-sorted-array.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) < 2:
            return len(nums)

        newArrayIndex = 0
        for nextIndex in range(1, len(nums)):
            if nums[newArrayIndex] != nums[nextIndex]:
                newArrayIndex += 1
                nums[newArrayIndex] = nums[nextIndex]
                
        return newArrayIndex + 1
